src/main/java/io/strimzi/kafka/bridge/mqtt/mapper/MqttKafkaRegexMapper.java

```diff
@@ -54,7 +57,22 @@ public MappingResult map(String mqttTopic) {
 
     @Override
     protected void buildOrCompilePatterns() {
-        this.rules.forEach(e -> patterns.add(Pattern.compile(e.getMqttTopicPattern())));
+        for (MappingRule rule : this.rules) {
+            int lastSlashIndex = rule.getMqttTopicPattern().length() - 3;
+            if (rule.getMqttTopicPattern().endsWith(PATTERN_MULTI_LEVEL_WILDCARD_REGEX) && rule.getMqttTopicPattern().split(MQTT_TOPIC_SEPARATOR).length > 1 && rule.getMqttTopicPattern().charAt(lastSlashIndex) == MQTT_TOPIC_SEPARATOR.charAt(0)){
+                // remove /.* from the end of the pattern
+                String regex = rule.getMqttTopicPattern().substring(0, lastSlashIndex) +
+                        // add the wildcard regex
+                        MqttKafkaRegexMapper.WILDCARD_REGEX;
+                this.patterns.add(Pattern.compile(regex));
+            } else if(rule.getMqttTopicPattern().endsWith("("+PATTERN_MULTI_LEVEL_WILDCARD_REGEX+")")){
+                throw new IllegalArgumentException("The pattern " + rule.getMqttTopicPattern() + " is not valid. You should not use .* in capture groups.\n" +
+                        "Example of correct use of .* are: building.*, building/.*, and etc.\n"+
+                        "Please refer to the documentation for more information.");
+            } else {
+                this.patterns.add(Pattern.compile(rule.getMqttTopicPattern()));
+            }
+        }
```

Comment:
I know you went further after our offline discussion but now I see this code adding more complexity for no additional value imho.
I think that we should revert back, by leaving the user the freedom to set a regex which we can just compile here (as the code was before).
Documenting that `buildings.*` and `buildings/.*` are two different things is important anyway.
Wdyt @antonio-pedro99 @kyguy ?

Reply:
maybe I should make the code cleaner?
I am apologist that the users should use `buildings/.*` to mean `buildings/#`
The previous implementation would not make this behaves exactly how it should behave when the topic is `buildings/#`

On the other hand, `building.*` looks more like `building#` (which is not a valid topic) and it would be confusing to the users.

And last but not least, if you noticed was replacing `.*` with `(?:\\/.*)?$` this the correct regex to deal with, because even when we use something like `building(?:/.*)?$`, `buildingslkmnskmskms` will never match the expression. However, this expression seems to big :( 

If you are fine, maybe we can go ahead and use directly `(?:/.*)?$` instead of `.*`

Reply:
I don't think defending against misuse of regex is worth the time or complexity for this PR. Maybe it is something we could revisit after mapping out all possible cases of misuse.

For now, I think we should compile the regex provided from the user directly as Paolo suggested, leaving the regex as a responsibility to the user. That was the original purpose of the regex mapper anyway right? To give users flexibility and save the application from the complexity.

Maybe we could provide some examples of some common (proper) regex patterns to use and list some common regex patterns to avoid.